### PR TITLE
WW-5414 Always call afterInvocation even in case of exception

### PR DESCRIPTION
**Pull request by lukaszlenart**
Close [WW-5414](https://issues.apache.org/jira/browse/WW-5414)

### CODE REVIEW

File: core/src/main/java/org/apache/struts2/interceptor/exec/StrutsBackgroundProcess.java [hunk 1]
```diff
@@ -61,11 +66,17 @@ public BackgroundProcess prepare() {
...
-                    afterInvocation();
                 } catch (Exception e) {
+                    LOG.warn("Exception during invokeActionOnly() execution", e);
                     exception = e;
                 } finally {
-                  done = true;
+                    try {
+                        afterInvocation();
+                    } catch (Exception ex) {
+                        exception = ex;
```
Comment (kusalk) [inline core/src/main/java/org/apache/struts2/interceptor/exec/StrutsBackgroundProcess.java:76]: Hmm what if `beforeInvocation()` already threw an exception, do we want to overwrite that?

Comment (lukaszlenart) [reply core/src/main/java/org/apache/struts2/interceptor/exec/StrutsBackgroundProcess.java:76]: Yeah, I know, after adding additional `afterException` and maybe `beforeException` fields to keep this information

Comment (kusalk) [reply core/src/main/java/org/apache/struts2/interceptor/exec/StrutsBackgroundProcess.java:76]: If we're only going to record one, the first one is probably more relevant - I'll let you make the call

**Review by kusalk — APPROVED** [review]

File: core/src/main/java/org/apache/struts2/interceptor/exec/StrutsBackgroundProcess.java (hunk 1 above)
Comment (lukaszlenart) [reply core/src/main/java/org/apache/struts2/interceptor/exec/StrutsBackgroundProcess.java:75]: Right, let's keep it simple, I will set `exception` only if wasn't already set

**Pull request merged**